fog/view/server/src/router_request_handler.rs

```diff
@@ -161,27 +194,32 @@ where
             break;
         }
 
-        authenticate_view_stores(
-            enclave.clone(),
-            processed_shard_response_data.view_store_uris_for_authentication,
-            logger.clone(),
-        )
-        .await?;
-    }
-
-    let query_response = enclave
-        .collate_shard_query_responses(sealed_query, query_responses)
-        .map_err(|err| {
-            router_server_err_to_rpc_status(
-                "Query: shard response collation",
-                RouterServerError::Enclave(err),
+        let view_store_uris_for_authentication =
+            processed_shard_response_data.view_store_uris_for_authentication;
+        if !view_store_uris_for_authentication.is_empty() {
+            authenticate_view_stores(
```

Comment:
:question: Just to double check the behaviour.
When `mc_fog_types::view::MultiViewStoreQueryResponseStatus::AuthenticationError` is returned for one of the shards we'll enter here. If *any* of the stores needing authentication fails we error out.  
We intentionally don't consider authentication needs as a *attempt`.

It seems like the worst case for this loop is if somehow we get 1 shard with `AuthenticationError` and the other shards are `Unknown`, then each time through the loop one of the `Unknown` shards transitions to `AuthenticationError`.
So worst case we loop for number of shards + `RETRY_COUNT`.

> when it should just pass back what it was able to process with the returned SUCCESS responses.
I got a bit confused on that description. 
Based on this implementation we want to return SUCCESS for `NotReady`, leaving the (non existent value) out of the responses.

Reply:
Yes this is the intended behavior (and is what Fog Ledger Router implemented). The case you outline is quite unlikely.